### PR TITLE
Defined nthreads in the other shooting-problem constructor + improved README file

### PR DESCRIPTION
**Pull request by cmastalli**
Note that if we do not modify `nthreads` in the constructor, the code crashes. This happens if we use the constructor with an externally allocated data.

### CODE REVIEW

**Review by proyan — APPROVED** [review]
Looks good to me, thanks for the changes Carlos.

Could you also please add the following papers to the Publications list:

https://hal.archives-ouvertes.fr/hal-02993058/

https://ieeexplore.ieee.org/abstract/document/9224427

**Comment by cmastalli** [comment]
> Looks good to me, thanks for the changes Carlos.
> 
> Could you also please add the following papers to the Publications list:
> 
> https://hal.archives-ouvertes.fr/hal-02993058/
> 
> https://ieeexplore.ieee.org/abstract/document/9224427

Done!

File: PUBLICATIONS.md [hunk 1]
```diff
@@ -21,15 +30,31 @@ T. G. Lembono, C. Mastalli, P. Fernbach, N. Mansard and S. Calinon. [Learning Ho
...
 T. Corbères, T. Flayols, P.A. Léziart, R. Budhiraja, N. Mansard
 [Comparison of predictive controllers for locomotion and balance recovery of quadruped robots](https://hal.laas.fr/hal-03034022/file/icra20_corberes.pdf), IEEE International Conference on Robotics and Automation (ICRA), 2021
 
+<a id="9">[9]</a>
+S. Kleff, A. Meduri, R. Budhiraja, N. Mansard and L. Righetti.
+[High-Frequency Nonlinear Model Predictive Control of a Manipulator](https://hal.archives-ouvertes.fr/hal-02993058v2/document), IEEE International Conference on Robotics and Automation (ICRA), 2021
+
+&nbsp;
+
+*iii. Co-design and biomechanics*
```
Comment (jcarpent) [inline PUBLICATIONS.md:50]: I think you split this section

File: PUBLICATIONS.md [hunk 2]
```diff
@@ -10,7 +12,14 @@ C. Mastalli, W. Merkt, J. Marti-Saumell, H. Ferrolho, J. Sola, N. Mansard, S. Vi
...
 
-*Other contributions build on top of Crocoddyl:*
+&nbsp;
+
+## Other references
+Here, there is a list of other contributions that were possible thanks to **Crocoddyl**:
+
+&nbsp;
+
+*i. Optimal control and learning:*
```
Comment (jcarpent) [inline PUBLICATIONS.md:22]: Same here.

File: PUBLICATIONS.md (hunk 1 above)
Comment (cmastalli) [reply PUBLICATIONS.md]: done

File: PUBLICATIONS.md (hunk 2 above)
Comment (cmastalli) [reply PUBLICATIONS.md]: done

**Pull request merged**